pkg/generic/thriftidl_provider.go

```diff
@@ -58,18 +64,24 @@ func NewThriftFileProvider(path string, includeDirs ...string) (DescriptorProvid
 
 // NewThriftFileProviderWithDynamicGo create a ThriftIDLProvider with dynamicgo by given path and include dirs
 func NewThriftFileProviderWithDynamicGo(path string, includeDirs ...string) (DescriptorProvider, error) {
+	return NewThriftFileProviderWithDynamicgoWithOption(path, []ThriftIDLProviderOption{}, includeDirs...)
+}
+
+func NewThriftFileProviderWithDynamicgoWithOption(path string, opts []ThriftIDLProviderOption, includeDirs ...string) (DescriptorProvider, error) {
 	p := &thriftFileProvider{
 		svcs: make(chan *descriptor.ServiceDescriptor, 1), // unblock with buffered channel
 		opts: &ProviderOption{DynamicGoEnabled: true},
 	}
-
-	svc, err := newServiceDescriptorFromPath(path, includeDirs...)
+	tOpts := &thriftIDLProviderOptions{}
+	tOpts.apply(opts)
+	parseMode := getParseMode(tOpts)
+	svc, err := newServiceDescriptorFromPath(path, parseMode, includeDirs...)
 	if err != nil {
 		return nil, err
 	}
 
 	// ServiceDescriptor of dynamicgo
-	dOpts := dthrift.Options{EnableThriftBase: true, ParseServiceMode: meta.ParseServiceMode(thrift.DefaultParseMode())}
+	dOpts := dthrift.Options{EnableThriftBase: true, ParseServiceMode: meta.ParseServiceMode(parseMode)}
```

Comment:
@AsterDY The logic is in getParseMode(). Only when users set ParseMode using the option (not thrift.SetDefaultParseMode() = global change), we use parse mode users set from the option.
Otherwise, we'll use thrift.DefaultParseMode().

Reply:
I mean how to ensure dynamicgo's ParseServiceMode == kitex's parseMode? There is no documented behavior. Even if I change the dynamicgo's ParseServiceMode=1234 or anything else, this can also be compiled

Reply:
I see. I'll add the logic